code/datums/riding.dm

```diff
@@ -34,16 +32,29 @@
 /datum/riding/proc/force_dismount(mob/living/M)
 	ridden.unbuckle_mob(M)
 
-//Override this to set your vehicle's various pixel offsets
-//if they differ between directions, otherwise use the
-//generic variables
 /datum/riding/proc/handle_vehicle_offsets()
+	var/ridden_dir = "[ridden.dir]"
+	var/passindex = 0
 	if(ridden.has_buckled_mobs())
 		for(var/m in ridden.buckled_mobs)
+			passindex++
 			var/mob/living/buckled_mob = m
-			buckled_mob.setDir(ridden.dir)
-			buckled_mob.pixel_x = generic_pixel_x
-			buckled_mob.pixel_y = generic_pixel_y
+			var/list/offsets = get_offsets(passindex)
+			dir_loop:
```

Comment:
why is this inner loop named? doesn't really do anything(unless for some reason the inner break is breaking the outer loop?)